### PR TITLE
Add check for `prefect-client` version in prefect-client workflow

### PR DESCRIPTION
**Pull request by desertaxle**
Fixes an issue where the `__version__` attribute is not populating correctly. The root cause is that the `setup.py` that the `prefect-client` package uses doe not include a `cmdclass`. That `cmdclass` generates the `_version.py` file that encodes the current version when we build the library. It was likely left out because we delete files in CI, which leads to `versioneer` adding a `.dirty` to the generated version.

After trying several other approaches, I changed the version style that `versioneer` uses for version generation to `pep440-pre`, which ignores dirty changes. This isn't ideal since we lose information with this style, but it does fix the issue. We should consider alternative ways of building `prefect-client` that don't require such changes.

I also added a check to ensure that `prefect-client` correctly matches the version of `prefect` from this point on.

Closes https://github.com/PrefectHQ/prefect/issues/15620

### CODE REVIEW

**Review by cicdw — APPROVED** [review]
nice work, works for me

File: .github/workflows/markdown-tests.yaml [hunk 1]
```diff
@@ -95,7 +95,7 @@ jobs:
 
       - name: Start server
         run: |
-          prefect server start&
+          PREFECT_HOME=$(pwd) prefect server start&
```
Comment (zzstoatzz) [inline .github/workflows/markdown-tests.yaml:98]: aha nice catch

File: .github/workflows/prefect-client.yaml [hunk 2]
```diff
@@ -56,6 +58,10 @@ jobs:
         env:
           TMPDIR: ${{ env.TMPDIR }}
 
+      - name: (DEBUG) Print versioneer output
+        run: python -c "import versioneer; print(versioneer.get_version())"
+        working-directory: ${{ env.TMPDIR }}
```
Comment (zzstoatzz) [inline .github/workflows/prefect-client.yaml:63]: do we want to keep this?

**Review by zzstoatzz — APPROVED** [review]

File: .github/workflows/prefect-client.yaml (hunk 2 above)
Comment (desertaxle) [reply .github/workflows/prefect-client.yaml:63]: Nah I'll get rid of it

**Pull request merged**